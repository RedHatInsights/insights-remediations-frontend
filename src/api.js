import { API_BASE, SOURCES_BASE } from './config';

import axios from 'axios';
import { RemediationsApi, ResolutionsApi, RemediationsApiAxiosParamCreator } from '@redhat-cloud-services/remediations-client';
import { DefaultApi } from '@redhat-cloud-services/sources-client';

/*
 * TODO: replace these with generated clients
 */
import { doGet } from './Utilities/http';
import urijs from 'urijs';

function url (...args) {
    const url = urijs(API_BASE).segment('remediations');
    args.forEach(segment => url.segment(segment));
    return url;
}

export const getRemediationStatus = id => doGet(url(id, 'status'));
export function getHosts () {
    return doGet('/api/inventory/v1/hosts');
}
/*
 * end of TODO
 */

class HttpError extends Error {
    constructor(description) {
        super('Error communicating with the server');
        this.description = description;
    }
}

async function authInterceptor (config) {
    await window.insights.chrome.auth.getUser();
    return config;
}

function responseDataInterceptor (response) {
    if (response.data) {
        return { ...response.data, etag: response.headers.etag };
    }

    return response;
}

function interceptor401 (err) {
    if (!err) {
        return;
    }

    if (err.response && err.response.status === 401) {
        window.insights.chrome.auth.logout();
        return false;
    }

    throw err;
}

function errorInterceptor (err) {
    if (!err) {
        return;
    }

    if (err.response && err.response.data && err.response.data.errors && err.response.data.errors.length) {
        const error = err.response.data.errors[0];

        if (error.details && error.details.name) {
            throw new HttpError(`${error.title} (${error.details.name})`);
        }

        throw new HttpError(error.title);
    }

    throw err;
}

const instance = axios.create();
instance.interceptors.request.use(authInterceptor);
instance.interceptors.response.use(responseDataInterceptor);
instance.interceptors.response.use(null, interceptor401);
instance.interceptors.response.use(null, errorInterceptor);

export const remediations = new RemediationsApi(undefined, API_BASE, instance);
export const resolutions = new ResolutionsApi(undefined, API_BASE, instance);
export const sources = new DefaultApi(undefined, SOURCES_BASE, instance);

export function downloadPlaybook (id) {
    return new Promise((resolve, reject) => {
        const tab = window.open(API_BASE + new RemediationsApiAxiosParamCreator().getRemediationPlaybook(id).url);
        if (!tab) {
            return reject();
        }

        const handle = setInterval(() => {
            if (tab.closed) {
                clearInterval(handle);
                resolve();
            }
        }, 500);
    });
}

export function getIsReceptorConfigured() {
    return doGet(`${window.location.origin}/api/sources/v2.0/endpoints?filter[receptor_node][not_nil]`);
}
