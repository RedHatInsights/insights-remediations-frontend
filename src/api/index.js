import { API_BASE, SOURCES_BASE } from '../config';

import axios from 'axios';
import {
  RemediationsApi,
  ResolutionsApi,
  RemediationsApiAxiosParamCreator,
} from '@redhat-cloud-services/remediations-client';
import { DefaultApi } from '@redhat-cloud-services/sources-client';

/*
 * TODO: replace these with generated clients
 */
import { doGet } from '../Utilities/http';
import urijs from 'urijs';

function url(...args) {
  const url = urijs(API_BASE).segment('remediations');
  args.forEach((segment) => url.segment(segment));
  return url;
}

export const getRemediationStatus = (id) => doGet(url(id, 'status'));
export function getHosts() {
  return doGet('/api/inventory/v1/hosts');
}
/*
 * end of TODO
 */

class HttpError extends Error {
  constructor(description) {
    super('Error communicating with the server');
    this.description = description;
  }
}

async function authInterceptor(config) {
  return config;
}

function responseDataInterceptor(response) {
  if (response.data) {
    return { ...response.data, etag: response.headers.etag };
  }

  return response;
}

function interceptor401(err) {
  if (!err) {
    return;
  }

  if (err.response && err.response.status === 401) {
    return false;
  }

  throw err;
}

function errorInterceptor(err) {
  if (!err) {
    return;
  }

  if (
    err.response &&
    err.response.data &&
    err.response.data.errors &&
    err.response.data.errors.length
  ) {
    const error = err.response.data.errors[0];

    if (error.details && error.details.name) {
      throw new HttpError(`${error.title} (${error.details.name})`);
    }

    throw new HttpError(error.title);
  }

  throw err;
}

const instance = axios.create();
instance.interceptors.request.use(authInterceptor);
instance.interceptors.response.use(responseDataInterceptor);
instance.interceptors.response.use(null, interceptor401);
instance.interceptors.response.use(null, errorInterceptor);

export const remediations = new RemediationsApi(undefined, API_BASE, instance);
export const resolutions = new ResolutionsApi(undefined, API_BASE, instance);
export const sources = new DefaultApi(undefined, SOURCES_BASE, instance);

export function downloadPlaybook(selectedIds) {
  return new Promise((resolve, reject) => {
    const tab =
      selectedIds.length > 1
        ? new RemediationsApiAxiosParamCreator()
            .downloadPlaybooks(selectedIds)
            .then((value) => window.open(API_BASE + value.url))
        : new RemediationsApiAxiosParamCreator()
            .getRemediationPlaybook(selectedIds[0])
            .then((value) => window.open(API_BASE + value.url));

    if (!tab) {
      return reject();
    }

    const handle = setInterval(() => {
      if (tab.closed) {
        clearInterval(handle);
        resolve();
      }
    }, 500);
  });
}

export function getIsReceptorConfigured() {
  return doGet(
    `${window.location.origin}/api/sources/v2.0/endpoints?filter[receptor_node][not_nil]`
  );
}

export function deleteSystemsFromRemediation(systems, remediation) {
  return Promise.all(
    systems.flatMap((system) =>
      system.issues.map((issue) =>
        remediations.deleteRemediationIssueSystem(
          remediation.id,
          issue.id,
          system.id
        )
      )
    )
  );
}

function checkResponse(r) {
  if (!r.ok) {
    const error = new Error(`Unexpected response code ${r.status}`);
    error.statusCode = r.status;
    throw error;
  }

  return r;
}

function json(r) {
  checkResponse(r);
  return r.json();
}

const headers = Object.freeze({
  'Content-Type': 'application/json; charset=utf-8',
});

export function createRemediation(data) {
  const uri = new urijs(API_BASE).segment('remediations').toString();
  return fetch(uri, {
    headers,
    method: 'POST',
    credentials: 'include',
    body: JSON.stringify(data),
  }).then(json);
}

export function patchRemediation(id, data) {
  const uri = new urijs(API_BASE)
    .segment('remediations')
    .segment(id)
    .toString();
  return fetch(uri, {
    headers,
    credentials: 'include',
    method: 'PATCH',
    body: JSON.stringify(data),
  }).then(checkResponse);
}

export function getRemediations() {
  const uri = new urijs(API_BASE)
    .segment('remediations')
    .query({ limit: 200 })
    .toString();
  return fetch(uri, { credentials: 'include' }).then(json);
}

export function getRemediation(id) {
  const uri = new urijs(API_BASE)
    .segment('remediations')
    .segment(id)
    .toString();
  return fetch(uri, { credentials: 'include' }).then(json);
}

export function getResolutionsBatch(issues) {
  const uri = new urijs(API_BASE).segment('resolutions').toString();
  return fetch(uri, {
    headers,
    credentials: 'include',
    method: 'POST',
    body: JSON.stringify({ issues }),
  }).then(json);
}
