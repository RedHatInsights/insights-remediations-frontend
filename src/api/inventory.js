const INVENTORY_API_BASE = '/api/inventory/v1';

import { APIFactory } from '@redhat-cloud-services/javascript-clients-shared';
import axiosInstance from '@redhat-cloud-services/frontend-components-utilities/interceptors';
import * as endpoints from '@redhat-cloud-services/host-inventory-client';

export const inventoryApi = APIFactory(INVENTORY_API_BASE, endpoints, {
  axios: axiosInstance,
});

export const getHostsById = (systems, { page, perPage }) =>
  inventoryApi.apiHostGetHostById(systems, null, perPage, page);

export const getHosts = () => inventoryApi.apiHostGetHostList();
