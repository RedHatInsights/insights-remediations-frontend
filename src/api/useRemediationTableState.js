import { useDeepCompareMemo } from 'use-deep-compare';
import { useSerialisedTableState } from '../Frameworks/AsyncTableTools/AsyncTableTools/hooks/useTableState';

const useRemediationTableState = (useTableState, paramsOption) => {
  const serialisedTableState = useSerialisedTableState();
  const {
    filters: filterState,
    pagination: { offset, limit } = {},
    sort: sortBy,
  } = serialisedTableState || {};

  const params = useDeepCompareMemo(() => {
    if (!Array.isArray(paramsOption)) {
      const { filter: filterParam, ...paramsParam } = paramsOption || {};
      const sort = sortBy;

      return useTableState
        ? {
            ...paramsParam,
            limit,
            offset,
            sort,
            ...filterParam,
            ...filterState,
          }
        : paramsOption;
    } else {
      return paramsOption;
    }
  }, [useTableState, filterState, limit, offset, sortBy, paramsOption]);

  return { params, hasState: !!serialisedTableState };
};

export default useRemediationTableState;
