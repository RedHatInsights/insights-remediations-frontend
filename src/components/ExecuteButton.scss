.rem-c-connection-status { margin-right: var(--pf-global--spacer--xs); }
.rem-c-connection-status-error { color: var(--pf-chart-global--danger--Color--100); }