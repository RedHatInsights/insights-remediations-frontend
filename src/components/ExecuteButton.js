/* eslint-disable camelcase */
import React, { useState, useEffect } from 'react';

import PropTypes from 'prop-types';
import { downloadPlaybook } from '../api';
import { Button, Modal, TextContent, Text, TextVariants, Alert } from '@patternfly/react-core';
import { TableHeader, Table, TableBody, TableVariant } from '@patternfly/react-table';
import { CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
import { Skeleton } from '@redhat-cloud-services/frontend-components';
import './ExecuteButton.scss';

const styledConnectionStatus = (status) => ({
    connected: (
        <TextContent>
            <Text component={ TextVariants.p }>
                <CheckCircleIcon
                    className="ins-c-remediations-reboot-check-circle ins-c-remediations-connection-status"
                    aria-label="connection status" />
            Ready
            </Text>
        </TextContent>),
    disconnected: (
        <TextContent>
            <Text component={ TextVariants.p }>
                Connection issue
            </Text>
            <Text component={ TextVariants.small }>
                Receptor not responding
            </Text>
            <Button
                key="troubleshoot"
                // eslint-disable-next-line no-console
                variant='link' onClick={ () => console.log('TODO: add link') }>
                Troubleshoot
            </Button>
        </TextContent>),
    no_executor: (
        <TextContent>
            <Text component={ TextVariants.p }>

                Cannot remediate - Direct connection.
            </Text>
            <Text component={ TextVariants.small }>
                 Connect your systems to Satellite to automatically remediate.
            </Text>
            <Button
                key="download"
                // eslint-disable-next-line no-console
                variant='link' onClick={ () => console.log('TODO: add link') }>
               Learn how to connect
            </Button>
        </TextContent>),
    no_source: (<TextContent>
        <Text component={ TextVariants.p }>
            Cannot remediate - Satellite not configured
        </Text>
        <Text component={ TextVariants.small }>
           Satellite not registered for Playbook execution
        </Text>
        <Button
            key="register"
            // eslint-disable-next-line no-console
            variant='link' onClick={ () => console.log('TODO: add link') }>
            Learn how to register Satellite
        </Button>
    </TextContent>),
    no_receptor: (<TextContent>
        <Text component={ TextVariants.p }>
            <ExclamationCircleIcon
                className="ins-c-remediations-connection-status-error ins-c-remediations-connection-status"
                aria-label="connection status" />
            Cannot remediate - Receptor not configured
        </Text>
        <Text component={ TextVariants.small }>
            Configure Receptor to automatically remediate
        </Text>
        <Button
            key="configure"
            // eslint-disable-next-line no-console
            variant='link' onClick={ () => console.log('TODO: add link') }>
            Learn how to configure
        </Button>
    </TextContent>)
})[status];

const ExecuteButton = ({
    isLoading,
    isDisabled,
    data,
    getConnectionStatus,
    remediationId,
    issueCount,
    runRemediation,
    etag,
    remediationStatus,
    setEtag }) => {
    const [ open, setOpen ] = useState(false);
    const [ isUserEntitled, setIsUserEntitled ] = useState(false);
    const [ showRefreshMessage, setShowRefreshMessage ] = useState(false);
    const isEnabled = () => true || localStorage.getItem('remediations:fifi:debug') === 'true';
    const isDebug = () => localStorage.getItem('remediations:debug') === 'true';

    useEffect(() => {
        window.insights.chrome.auth.getUser().then(user => setIsUserEntitled(user.entitlements.smart_management.is_entitled));
    }, []);

    useEffect(() => {
        if (remediationStatus === 'changed') {
            getConnectionStatus(remediationId);
            setShowRefreshMessage(true);
        } else if (remediationStatus === 'fulfilled') {
            setOpen(false);
        }
    }, [ remediationStatus ]);

    const [ connected, disconnected ] = data.reduce(
        ([ pass, fail ], e) => (e.connection_status === 'connected' ? [ [ ...pass, e ], fail ] : [ pass, [ ...fail, e ] ])
        , [ [], [] ]
    );

    const rows = [ ...connected, ...disconnected ].map(con =>
        ({ cells: [
            con.executor_name || 'Direct connection',
            con.system_count,
            isUserEntitled && { title: styledConnectionStatus(con.connection_status) }
        ]})
    );
    const connectedCount = connected.reduce((acc, e) => e.system_count + acc, 0);
    const systemCount = data.reduce((acc, e) => e.system_count + acc, 0);

    const pluralize = (number, str) => number > 1 ? `${number} ${str}s` : `${number} ${str}`;
    return (isUserEntitled && isEnabled()
        ? <React.Fragment>
            <Button
                isDisabled={ isDisabled }
                onClick={ () => { setOpen(true); getConnectionStatus(remediationId); } }>
        Execute playbook
            </Button>
            <Modal
                className="ins-c-dialog"
                width={ '50%' }
                title={ 'Execute playbook' }
                isOpen={ open }
                onClose={ () => {
                    setShowRefreshMessage(false);
                    setOpen(false);
                } }
                isFooterLeftAligned
                actions={ [
                    <Button
                        key="confirm"
                        variant="primary"
                        isDisabled={ connected.length === 0 }
                        onClick={ () => { runRemediation(remediationId, etag); } }>
                        { isLoading ? 'Execute playbook' : `Execute playbook on ${pluralize(connectedCount, 'system')}` }
                    </Button>,
                    <Button
                        key="download"
                        variant='link' onClick={ () => downloadPlaybook(remediationId) }>
                        Download playbook
                    </Button>,
                    (isDebug()
                        ? <Button
                            key="reset-etag"
                            onClick={ () => setEtag('test') }>
                        Reset etag
                        </Button>
                        : null)
                ] }
            >
                <div>
                    { showRefreshMessage
                        ? <Alert variant="warning" isInline
                            title="The connection status of systems associated with this Playbook has changed. Please review again." />
                        : null }
                    <TextContent>
                        { isLoading
                            ? <Skeleton size='lg'/>
                            : <Text component={ TextVariants.p }>
                                Playbook contains <b>{ `${pluralize(issueCount, 'issue')}` }</b> affecting
                                <b>  { `${pluralize(systemCount, 'system')}.` } </b>
                            </Text> }
                        <Text component={ TextVariants.p }>
                        Systems connected to a Satellite instance and configured with Receptor can be automatically remediated.
                            To remediate other systems, download the Ansible Playbook.

                        </Text>
                        <Text component={ TextVariants.h4 }>Connection status of systems</Text>
                    </TextContent>
                    { isLoading
                        ? <Skeleton size='lg'/>
                        : <Table
                            variant={ TableVariant.compact }
                            aria-label="Systems"
                            cells={ [{
                                title: 'Connection type', value: 'type'
                            }, {
                                title: 'Systems', value: 'count'
                            }, isUserEntitled && {
                                title: 'Connection status', value: 'status'
                            } ] }
                            rows={ rows }
                        >
                            <TableHeader />
                            <TableBody />
                        </Table> }
                </div>
            </Modal>
        </React.Fragment>
        : null
    );
};

ExecuteButton.propTypes = {
    isLoading: PropTypes.bool,
    data: PropTypes.array,
    getConnectionStatus: PropTypes.func,
    runRemediation: PropTypes.func,
    remediationId: PropTypes.string,
    remediationStatus: PropTypes.string,
    issueCount: PropTypes.number,
    etag: PropTypes.string,
    setEtag: PropTypes.func,
    isDisabled: PropTypes.bool
};

ExecuteButton.defaultProps = {
    data: [],
    isDisabled: false
};

export default ExecuteButton;
