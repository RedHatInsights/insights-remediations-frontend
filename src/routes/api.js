import { remediationsApi } from '../api';

export const getRemediationDetails = ({ remId }) =>
  remediationsApi.getRemediation(remId);

export const getRemediations = (params) =>
  remediationsApi.getRemediations(params);

export const getRemediationPlaybook = ({ remId }) =>
  remediationsApi.listPlaybookRuns(remId);

export const checkExecutableStatus = ({ remId }) =>
  remediationsApi.checkExecutable(remId);

export const getRemediationPlaybookSystemsList = ({
  remId,
  playbook_run_id,
}) => {
  return remediationsApi.getPlaybookRunSystems(remId, playbook_run_id);
};

export const getPlaybookLogs = (params) =>
  remediationsApi.getPlaybookRunSystemDetails(
    params.remId,
    params.playbook_run_id,
    params.system_id,
  );

export const getRemediationsList = () =>
  remediationsApi.getRemediations({ fieldsData: ['name'] });

export const updateRemediationPlans = (params) => {
  const { id, ...updateData } = params;
  return remediationsApi.updateRemediation(id, updateData);
};

export const deleteRemediation = ({ id }) =>
  remediationsApi.deleteRemediation(id);

export const deleteRemediationList = ({ remediation_ids }) =>
  remediationsApi.deleteRemediations({
    remediation_ids,
  });

export const executeRemediation = ({ id, etag, exclude }) => {
  remediationsApi.runRemediation(
    id,
    { exclude },
    {
      headers: { 'If-Match': etag },
    },
  );
};

export const API_BASE = '/api/remediations/v1';
