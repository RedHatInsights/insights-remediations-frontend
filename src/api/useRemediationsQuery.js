import { useCallback } from 'react';
// import useRemediationsApi from '../Utilities/Hooks/api/useRemediationsApi';
import useQuery from '../Utilities/Hooks/useQuery';
import useRemediationTableState from './useRemediationTableState';
import useFetchTotalBatched from '../Utilities/Hooks/useFetchTotalBatched';

/**
 *  @typedef {object} useRemediationQueryParams
 *
 *  @property {string} [filter]            Scoped search filter string for the endpoint
 *  @property {object} [pagination]        API pagination params
 *  @property {object} [pagination.offset] Pagination offset
 *  @property {object} [pagination.limit]  Pagination limit (maximum 100)
 *  @property {object} [sortBy]            SortBy string for the API (usually 'attribute:desc')
 *
 */

/**
 *
 * Hook to use a Remediation REST API v2 endpoint with useQuery.
 * Optionally support for using the serialised table state if a `<TableStateProvider/>` is available.
 *
 *  @param   {Function}                 endpoint                String of the javascript-clients export for the needed endpoint
 *
 *  @param   {object}                   [options]               Options for useRemediationsQuery & useQuery
 *  @param   {useRemediationQueryParams} [options.params]        API endpoint params
 *  @param   {boolean}                  [options.useTableState] Use the serialised table state
 *
 *  @returns {useQueryReturn}                                   An object containing a data, loading and error state, as well as a fetch and refetch function.
 *
 *  @category Remediation
 *  @subcategory Hooks
 *
 * @example
 *
 */
const useRemediationsQuery = (
  endpoint,
  {
    params: paramsOption,
    useTableState = false,
    batched = false,
    skip: skipOption,
    batch = {},
    ...options
  } = {}
) => {
  // const apiEndpoint = useRemediationsAPI(endpoint);
  const { params, hasState } = useRemediationTableState(
    useTableState,
    paramsOption
  );
  const skip = !!(useTableState && !hasState) || !!skipOption;

  const {
    data: queryData,
    error: queryError,
    loading: queryLoading,
    fetch: queryFetch,
    meta: metaData,
  } = useQuery(endpoint, {
    skip: batched ? true : skip,
    ...options,
    params,
  });

  const fetchForBatch = useCallback(
    async (offset, limit, params) =>
      await queryFetch({ limit, offset, ...params }, false),
    [queryFetch]
  );

  const {
    loading: batchedLoading,
    data: batchedData,
    error: batchedError,
    fetch: batchedFetch,
  } = useFetchTotalBatched(fetchForBatch, {
    skip: !batched ? true : skip,
    ...batch,
  });

  const exporter = async () => (await batchedFetch()).data;

  const fetchAllIds = async () =>
    (await batchedFetch()).data.map(({ id }) => id);

  return {
    ...(batched
      ? {
          data: batchedData,
          error: batchedError,
          loading: batchedLoading,
        }
      : {
          data: queryData,
          error: queryError,
          loading: queryLoading,
          meta: metaData,
        }),
    fetch: queryFetch,
    fetchBatched: batchedFetch,
    fetchAllIds,
    exporter,
  };
};
export default useRemediationsQuery;
