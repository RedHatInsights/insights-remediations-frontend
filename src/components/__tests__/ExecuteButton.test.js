/* eslint-disable camelcase */
import React from 'react';
import { mount } from 'enzyme';
import ExecuteButton from '../ExecuteButton';
import toJson from 'enzyme-to-json';
import * as api from '../../api';

api.downloadPlaybook = jest.fn();
global.insights = { chrome: { auth: { getUser: () => new Promise(() => ({ entitlements: { smart_management: { isEntitled: true }}})) }}};

const data = [
    { executor_id: '722ec903-f4b5-4b1f-9c2f-23fc7b0ba390', executor_type: 'satellite', executor_name: 'Satellite 1 (connected)',
        system_count: 1, connection_status: 'connected' },
    { executor_id: null, executor_type: null, executor_name: null, system_count: 4, connection_status: 'no_executor' }
];

describe('Execute button', () => {

    test('fullfiled request', () => {
        const tree = mount(<ExecuteButton
            data={ data }
            isLoading
            status={ 'fullfiled' }
            issueCount={ 1 }
            remediationId='id'
            isDisabled={ false }
            getEndpoint={ () => null }
            sources={ { data: {}} }
            getConnectionStatus={ () => null }/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

    test('pending request', () => {
        const tree = mount(<ExecuteButton
            data={ [] }
            isLoading
            status={ 'pending' }
            issueCount={ 1 }
            remediationId='id'
            isDisabled={ false }
            sources={ { data: {}} }
            getConnectionStatus={ () => null } />);
        expect(toJson(tree)).toMatchSnapshot();
    });
});
