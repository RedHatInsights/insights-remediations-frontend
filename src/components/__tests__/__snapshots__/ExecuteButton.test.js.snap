// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Execute button fullfiled request 1`] = `
<ExecuteButton
  data={
    Array [
      Object {
        "connection_status": "connected",
        "executor_id": "722ec903-f4b5-4b1f-9c2f-23fc7b0ba390",
        "executor_name": "Satellite 1 (connected)",
        "executor_type": "satellite",
        "system_count": 1,
      },
      Object {
        "connection_status": "no_executor",
        "executor_id": null,
        "executor_name": null,
        "executor_type": null,
        "system_count": 4,
      },
    ]
  }
  getConnectionStatus={[Function]}
  getEndpoint={[Function]}
  isDisabled={false}
  isLoading={true}
  issueCount={1}
  remediationId="id"
  sources={
    Object {
      "data": Object {},
    }
  }
  status="fullfiled"
/>
`;

exports[`Execute button pending request 1`] = `
<ExecuteButton
  data={Array []}
  getConnectionStatus={[Function]}
  isDisabled={false}
  isLoading={true}
  issueCount={1}
  remediationId="id"
  sources={
    Object {
      "data": Object {},
    }
  }
  status="pending"
/>
`;
